screen2.0/src/app/search/MainResultsFilters.tsx

```diff
@@ -0,0 +1,767 @@
+
+  const [gene, setGene] = useState<string>("")
+
+
+  const {
+    data: geneTranscripts
+  } = useQuery(GENE_TRANSCRIPTS_QUERY, {
+    variables: {
+      assembly: props.mainQueryParams.coordinates.assembly.toLowerCase(),
+      name: [props.mainQueryParams.gene.name && props.mainQueryParams.gene.name.toUpperCase()]
+    },
+    skip: !props.mainQueryParams.gene.name,
+    fetchPolicy: "cache-and-network",
+    nextFetchPolicy: "cache-first"
+
+  })
+
+  //Recalculate when geneTranscripts available, near
+  useEffect(() => {
+    if (geneTranscripts?.gene?.length > 0) {
+      const TSSs = geneTranscripts.gene[0].transcripts.map(t => {
+        if (geneTranscripts.gene[0].strand === "+") {
+          return t.coordinates.start
+        } else {
+          return t.coordinates.end
+        }
+      })
+      console.log("Max" + Math.max(...TSSs))
+      console.log("Min" + Math.min(...TSSs))
+      props.setTSSs(TSSs)
+    }
+  }, [geneTranscripts])
+
+  //Recalculate TSS ranges when gene distance changes
+  useEffect(() => {
+    const TSSranges: { start: number, end: number }[] = props.TSSs?.map((tss) => {
+      return { start: Math.max(0, tss - props.mainQueryParams.gene.distance), end: tss + props.mainQueryParams.gene.distance }
+    })
+    props.setTSSranges(TSSranges)
+  }, [props.mainQueryParams.gene.distance])
+  
+  function valuetext(value: number) {
+    return `${value}kb`;
+  }
+
+  const filteredBiosamples: FilteredBiosampleData = useMemo(() => {
+    if (props.byCellType) {
+      return (
+        filterBiosamples(
+          parseByCellType(props.byCellType),
+          props.biosampleTableFilters.Tissue,
+          props.biosampleTableFilters.PrimaryCell,
+          props.biosampleTableFilters.CellLine,
+          props.biosampleTableFilters.InVitro,
+          props.biosampleTableFilters.Organoid
+        )
+      )
+    } else return []
+  }, [props.byCellType, props.biosampleTableFilters.Tissue, props.biosampleTableFilters.PrimaryCell, props.biosampleTableFilters.CellLine, props.biosampleTableFilters.InVitro, props.biosampleTableFilters.Organoid])
+
+  //This could be refactored to improve performance in SNP/Gene filters. The onRowClick for each table depends on setting main query params, which the gene/snp filters also modify
+  //This is recalculated every time those sliders are moved.
+  const biosampleTables = useMemo(() => {
+      const cols = [
+        {
+          header: "Biosample",
+          value: (row) => row.summaryName,
+          render: (row) => (
+            <Tooltip title={"Biosample Type: " + row.biosampleType} arrow>
+              <Typography variant="body2">{row.summaryName}</Typography>
+            </Tooltip>
+          ),
+        },
+        {
+          header: "Assays",
+          value: (row) => Object.keys(row.assays).filter((key) => row.assays[key] === true).length,
+          render: (row) => {
+            const fifth = (2 * 3.1416 * 10) / 5
```

Comment:
I can we can rename fifth variable here .

Reply:
Changed to "fifthOfCircle" to try to be more clear. Open to other names